src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapMetadata.cs

```diff
@@ -437,22 +437,22 @@ UInt32 index
                     return (int) WinCodecErrors.WINCODEC_ERR_UNSUPPORTEDOPERATION;
                 }
 
-                _metadataBlocks.Remove(index);
```

Comment:
Basically previously it would look for the index of the writer, as if it was an item (which would never succeed), the correct implementation is to remove the writer at the index.